src/main/java/http_server/ResponseBuilder.java

```diff
@@ -0,0 +1,119 @@
+package http_server;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ResponseBuilder {
+    private Response response;
+    private Request request;
+    private Router router;
+    private HashMap<String, Callback> methodCollection;
+    private String crlf = "\r\n";
+
+    public ResponseBuilder(Response response, Request request, Router router) {
+        this.response = response;
+        this.request = request;
+        this.router = router;
+        this.methodCollection = router.getMethodCollection(request.getRoute());
+    }
+
+    public String build() {
+        if (methodCollection.isEmpty()) {
+            response.setStatus(StatusCode.NOT_FOUND.get());
```

Comment:
I think using enums for status codes is an interesting choice, but I'm not sure I see the value in it over simple constants based on how you use them. Jayden's review will elaborate.